src/dex/inception/inception.ts

```diff
@@ -0,0 +1,238 @@
+import { Interface, JsonFragment } from '@ethersproject/abi';
+import { AsyncOrSync } from 'ts-essentials';
+import { NumberAsString, SwapSide } from '@paraswap/core';
+import {
+  AdapterExchangeParam,
+  Address,
+  DexExchangeParam,
+  ExchangePrices,
+  Logger,
+  PoolLiquidity,
+  PoolPrices,
+  Token,
+} from '../../types';
+import INCEPTION_ABI from '../../abi/inception/inception-vault.json';
+import INCEPTION_POOL_ABI from '../../abi/inception/inception-ineth-pool.json';
+import { Network, NULL_ADDRESS } from '../../constants';
+import * as CALLDATA_GAS_COST from '../../calldata-gas-cost';
+import { getDexKeysWithNetwork, Utils } from '../../utils';
+import { IDex } from '../../dex/idex';
+import { IDexHelper } from '../../dex-helper/idex-helper';
+import { DexParams, InceptionDexData, PoolState } from './types';
+import { SimpleExchange } from '../simple-exchange';
+import { InceptionConfig, InceptionPricePoolConfig } from './config';
+import { getTokenFromAddress, setTokensOnNetwork, Tokens } from './tokens';
+import { getOnChainState, getTokenList } from './utils';
+import { InceptionEventPool } from './inception-event-pool';
+
+const DECIMALS = BigInt(1e18);
+
+export class Inception
+  extends SimpleExchange
+  implements IDex<InceptionDexData>
+{
+  protected vaultInterface: Interface;
+  protected poolInterface: Interface;
+
+  readonly hasConstantPriceLargeAmounts = false;
+  readonly needWrapNative = false;
+  readonly isFeeOnTransferSupported = false;
```

Comment:
Please check this [update](https://github.com/paraswap/paraswap-dex-lib/pull/733/commits/3ba1a59a4b244a0858960c1416e25be09f129733) to enable swaping stETH (that is marked as fee token due to rounding)